pkg/backend/filestate/lock.go

```diff
@@ -0,0 +1,157 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package filestate
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"github.com/pulumi/pulumi/pkg/backend"
+	"github.com/pulumi/pulumi/pkg/util/logging"
+	"os"
+	"os/user"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+type lockContent struct {
+	Pid       int       `json:"pid"`
+	Username  string    `json:"username"`
+	Hostname  string    `json:"hostname"`
+	Timestamp time.Time `json:"timestamp"`
```

Comment:
looked, but couldn't see 'Timestamp' being used.  necessary?  If not, feel free to remove.

Reply:
I included the timestamp incase we wanted locks to expire or have some kind of freshness to them. Rather than taking it out I think it would be better to display the lock age when we tell the user about the existing lock.